### PR TITLE
Respect subcommittees in gossip validations for Altair

### LINKED ISSUE #2324: SignedContributionAndProof gossip validation does not check validator is in subcommittee
The honest validator spec says that each slot members of each subcommittee are selected to aggregate (https://github.com/ethereum/eth2.0-specs/blob/dev/specs/altair/validator.md#sync-committee-contributions).  My understanding is that validators would only check if they are an aggregator for the subcommittees they are in, not all subcommittees.

But the gossip validation only checks that the aggregator's validator index is within the current sync committee, not the specific subcommittee (https://github.com/ethereum/eth2.0-specs/blob/dev/specs/altair/p2p-interface.md#sync_committee_contribution_and_proof):
>[REJECT] The aggregator's validator index is within the current sync committee -- i.e. state.validators[contribution_and_proof.aggregator_index].pubkey in state.current_sync_committee.pubkeys.

I believe it should be checking that the aggregator's pub key is in the sublist of `state.current_sync_committee.pubkeys` that pertains to `contribution.subcommittee_index`.  So something like `state.current_sync_commitee.pubkeys[subcommitteeIndex * (SYNC_COMMITTEE_SIZE // SYNC_COMMITTEE_SUBNET_COUNT):(subcommitteeIndex + 1) * (SYNC_COMMITTEE_SIZE // SYNC_COMMITTEE_SUBNET_COUNT)` and I really hope there's a nicer way to write that in python.

### PR DESCRIPTION
**Pull request by ralexstokes**
Fixes #2324. Along with some other minor cleanups.

### CODE REVIEW

File: specs/altair/p2p-interface.md [hunk 1]
```diff
@@ -83,18 +83,26 @@ See the [state transition document](./beacon-chain.md#beaconblockbody) for Altai
  
 This topic is used to propagate partially aggregated sync committee signatures to be included in future blocks.
 
-The following validations MUST pass before forwarding the `signed_contribution_and_proof` on the network; define `contribution_and_proof = signed_contribution_and_proof.message` and `contribution = contribution_and_proof.contribution` for convenience:
+The following validations MUST pass before forwarding the `signed_contribution_and_proof` on the network; define `contribution_and_proof = signed_contribution_and_proof.message`, `contribution = contribution_and_proof.contribution`, and the following function `get_sync_subcommittee_pubkeys` for convenience:
+
+```python
+SYNC_SUBCOMMITTEE_SIZE = SYNC_COMMITTEE_SIZE // SYNC_COMMITTEE_SUBNET_COUNT
+
+def get_sync_subcommittee_pubkeys(state: BeaconState, subcommittee_index: uint64) -> Sequence[BLSPubkey]:
```
Comment (djrtwo) [inline specs/altair/p2p-interface.md:91]: Does p2p spec get built with python? I don't suppose so

Either toss this in validator.md or add p2p-interface.md into the build

File: specs/altair/p2p-interface.md [hunk 2]
```diff
@@ -109,8 +117,8 @@ The following validations MUST pass before forwarding the `sync_committee_signat
 - _[IGNORE]_ The signature's slot is for the current slot, i.e. `sync_committee_signature.slot == current_slot`.
 - _[IGNORE]_ The block being signed over (`sync_committee_signature.beacon_block_root`) has been seen (via both gossip and non-gossip sources).
 - _[IGNORE]_ There has been no other valid sync committee signature for the declared `slot` for the validator referenced by `sync_committee_signature.validator_index`.
-- _[REJECT]_ The validator producing this `sync_committee_signature` is in the current sync committee, i.e. `state.validators[sync_committee_signature.validator_index].pubkey in state.current_sync_committee.pubkeys`.
-- _[REJECT]_ The `subnet_id` is correct, i.e. `subnet_id in compute_subnets_for_sync_committee(state, sync_committee_signature.validator_index)`.
+- _[REJECT]_ The `subnet_id` is correct, i.e. `subnet_id in compute_subnets_for_sync_committee(state, sync_committee_signature.validator_index)`.
```
Comment (djrtwo) [inline specs/altair/p2p-interface.md:120]: Slight suggestion on language to imply that there might be multiple valid things here and it's wrt the validator
```suggestion
- _[REJECT]_ The `subnet_id` is valid for the given validator, i.e. `subnet_id in compute_subnets_for_sync_committee(state, sync_committee_signature.validator_index)`. 
```

**Review by djrtwo — COMMENTED** [review]
cool! a few minor comments

File: specs/altair/p2p-interface.md (hunk 1 above)
Comment (ralexstokes) [reply specs/altair/p2p-interface.md:89]: makes more sense to scan `p2p-interface.md` but looks like we dont currently support constants in markdown blocks so had to fold it in

**Comment by ralexstokes** [comment]
update: this should be ready pending further review -- the CI test failure is related to a parse error from the executable spec library; i'm leaning towards throwing up another (blocking) PR to handle the parsing error before getting this to a place where the CI passes

**Comment by djrtwo** [comment]
Looks good.

So keep this PR as is until another PR to fix build?

**Comment by ralexstokes** [comment]
changes here should fix this PR as is: https://github.com/ethereum/eth2.0-specs/pull/2345

**Review by djrtwo — APPROVED** [review]
great work!

**Pull request merged**